### PR TITLE
Deploy erc20 contract as first transaction

### PR DESCRIPTION
**Pull request by rishflab**
Context 

"One thing I noticed when working on the specify-addresses branch: At some point the ERC20 contract had a different address. That most likely means that it was deployed with a different nonce than before. That could be due to the Ethereum transactions we do.
Would it be possible to deploy the ERC20 contract as first transaction, so it has a deterministic address?" ~ @da-kami

### CODE REVIEW

**Comment by rishflab** [comment]
Holding off on merge, doesnt seem to fund DAI

File: scripts/src/docker/ethereum.rs [hunk 1]
```diff
@@ -38,6 +37,80 @@ pub struct GethInstance {
...
+        Ok(())
+    }
+
+    async fn fund_erc20(&self, address: Address, contract_address: Address) -> anyhow::Result<()> {
+        send_transaction(
+            self.client.clone(),
+            Some(contract_address),
+            100_000,
+            U256::from(0u64),
+            Vec::new(),
```
Comment (thomaseizinger) [inline scripts/src/docker/ethereum.rs:80]: You are not passing the correct data here for funding ERC20. Check your diff and see how the old code put `transfer_fn` here.

**Comment by rishflab** [comment]
Tested this PR. The erc20 contract does not get a deterministic address" by deploying it before funding ethereum to the accounts. The only thing useful here is the refactor which makes the ethereum setup easier to understand and I increased the default amounts.

**Comment by thomaseizinger** [comment]
> Tested this PR. The erc20 contract does not get a deterministic address" by deploying it before funding ethereum to the accounts. The only thing useful here is the refactor which makes the ethereum setup easier to understand and I increased the default amounts.

The contract is not deterministic because you are deploying it with the dev account and that seems to change every time.
To avoid this, you can create yourself a static dev account, fund that from the dynamic one and deploy the contract from that address :)

**Comment by da-kami** [comment]
Here is the ticket, do later: https://github.com/comit-network/create-comit-app/issues/825

**Pull request closed without merging**